### ISSUE TITLE
Applying a lambda map with an empty range of parameters

### ISSUE THREAD

**Issue by wwitzel** [PyProveIt/Prove-It#174]
@jmadridlarra reported the following error (in Issue #170 ):
InstantiationFailure: Proof step failed assuming {n in Naturals}:
Attempting to instantiate |- forall_{n in Naturals} [forall_{f_{1}, ..f_{_a}.., f_{n}, i | i in Naturals} (|(f_{1}(1), ..f_{1}(x).., f_{1}(i), ....f_{k}(1), ..f_{k}(x).., f_{k}(i)...., f_{n}(1), ..f_{n}(x).., f_{n}(i))| \\ = (n * i))] with {n: 0, f: (j -> a_{1,j}, ..j -> a_{i,j}.., j -> a_{0,j}), i: n, (f_{1}, ..f_{_a}.., f_{0}): (j -> a_{1,j}, ..j -> a_{i,j}.., j -> a_{0,j})}:
Failure to apply (n, f_{1}, ..f_{_a}.., f_{0}, i) -> (i in Naturals) to (0, [j -> a_{1,j}], ..[j -> a_{i,j}].., [j -> a_{0,j}], n) assuming (n in Naturals,):
<bound method Len.doReducedEvaluation of |n|> must return an KnownTruth equality with an irreducible value on the right side, not |- |n| = |n| for |n| assuming (n in Naturals,)

Basically, Prove-It is getting confused in handling an emtpy range of parameters to be matched with zero operands.  We either need to handle this proper, or give an appropriate error message that we do not handle a Lambda application with an empty range of parameters.  Is it ever valid to do this?  In the above example, it turned out to not really be what we wanted to do.  But maybe there are instances where it is valid and we should properly handle this case.

**Comment by wwitzel** [comment]
I believe this is addressed via the 'param_to_num_operand_entries' optional argument.
